src/optimization/Optimizer.cpp

```diff
@@ -140,9 +140,11 @@ const OptimizationPass optimizations::SPLIT_READ_WRITES = OptimizationPass("Spli
 const OptimizationPass optimizations::REORDER = OptimizationPass("ReorderInstructions", reorderWithinBasicBlocks, 140);
 const OptimizationPass optimizations::COMBINE = OptimizationPass("CombineALUIinstructions", combineOperations, 150);
 const OptimizationPass optimizations::UNROLL_WORK_GROUPS = OptimizationPass("UnrollWorkGroups", unrollWorkGroups, 160);
+const OptimizationPass optimizations::INSTRUCTION_SCHEDULING = OptimizationPass("InstructionScheduling", instructionScheduling, 170);
```

Comment:
You probably should move the instruction-scheduling before unrolling the work-groups, unless you included the work-group loop (repeating the complete code) on purpose.